packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -281,7 +281,7 @@ Future<void> _runBuild(
 
   // MSBuild sends all output to stdout, including build errors. This surfaces
   // known error patterns.
-  final RegExp errorMatcher = RegExp(r':\s*(?:warning|(?:fatal )?error).*?:');
+  final RegExp errorMatcher = RegExp(r':\s*(?:warning|(?:fatal )?error).*?:|.*Error detected in pubspec.yaml:.*|.*No file or variants found for asset: .*');
```

Comment:
Having to encode the specific error message is very unfortunate; any other pubspec error is going to be missing all of the actual details until we add each one as it comes up. But without fundamentally reworking `stream` I don't have a better suggestion.

We should at least make this more maintainable and clearly documented though. Something like:
```
final List<String> errorMessages = <String>[
  r':\s*(?:warning|(?:fatal )?error).*?:',
  r'Error detected in pubspec.yaml:',
];
// Known secondary error lines for pubspec.yaml errors:
final List<String> pubspecErrorDetails = <String>[
  r'No file or variants found for asset:',
];
final RegExp errorMatcher = RegExp(<String>[
  ...errorMessages,
  ...pubspecErrorDetails,
].join('|'));
```

(Note that I have remove the leading and trailing `.*` from your strings. As you can see from the existing regex, this is not a full-line match.)